### PR TITLE
Bugfix 692

### PR DESCRIPTION
**Pull request by idanov**
This PR fixes:
 - `SexpList` apply should have a tail recursive instead of recurse that could blow up the stack
 - option not to write out case class fields that are `nil`

These can be found in issue #692 
@iyadi @konradwudkowski

### CODE REVIEW

File: s-express/src/test/scala/org/ensime/sexp/SexpSpec.scala [hunk 1]
```diff
@@ -40,6 +42,12 @@ class SexpSpec extends EnsimeSpec {
     }
   }
 
+  it should "support a list with 1000000 elements" in {
+    val data = List.fill(1000000)(SexpChar('a'))
+    val res = Try(SexpList(data))
+    res shouldBe a[Success[Sexp]]
```
Comment (fommil) [inline s-express/src/test/scala/org/ensime/sexp/SexpSpec.scala]: the CI doesn't like this line due to erasure. maybe do a pattern match on the Try?

**Comment by fommil** [comment]
I'll close this and raise a PR with your test slightly tweaked.

**Comment by fommil** [comment]
thanks! Merged in a different PR where I made the one line fix.

**Comment by idanov** [comment]
@fommil great :)

**Pull request closed without merging**